src/Acuminator/Acuminator.Vsix/Commands/DiagnosticSuppression/RoslynDiagnosticService.cs

```diff
@@ -0,0 +1,178 @@
+		public Type DiagnosticAnalyzerServiceType { get; }
+
+		public Type DiagnosticDataType { get; }
+
+		private readonly object _roslynAnalyzersService;
+		private readonly MethodInfo _getDiagnosticOnTextSpanMethod;
+		private readonly PropertyInfo _taskResultPropertyInfo;
+
+		/// <summary>
+		/// Gets the instance of the service.
+		/// </summary>
+		public static RoslynDiagnosticService Instance
+		{
+			get;
+			private set;
+		}
+
+		/// <summary>
+		/// Initializes the singleton instance of the command.
+		/// </summary>
+		/// <param name="package">Owner package, not null.</param>
+		public static void Initialize(Package package)
+		{
+			if (Interlocked.CompareExchange(ref _isServiceInitialized, value: INITIALIZED, comparand: NOT_INITIALIZED) == NOT_INITIALIZED)
+			{
+				var componentModel = package?.GetService<SComponentModel, IComponentModel>();
+
+				Instance = new RoslynDiagnosticService(package, componentModel);
+			}
+		}
+
+		private RoslynDiagnosticService(Package package, IComponentModel componentModel)
+		{
+			package.ThrowOnNull(nameof(package));
+			componentModel.ThrowOnNull(nameof(componentModel));
+
+			_package = package;
+			_componentModel = componentModel;
+
+			DiagnosticAnalyzerServiceType = GetInternalRoslynServiceType();
+			DiagnosticDataType = GetDiagnosticDataType();
+			DiagnosticAnalyzerServiceType.ThrowOnNull(nameof(DiagnosticAnalyzerServiceType));
+			DiagnosticDataType.ThrowOnNull(nameof(DiagnosticDataType));
+
+			_roslynAnalyzersService = GetDiagnosticServiceInstance(componentModel, DiagnosticAnalyzerServiceType);
+			_roslynAnalyzersService.ThrowOnNull(nameof(_roslynAnalyzersService));
+
+			_getDiagnosticOnTextSpanMethod = DiagnosticAnalyzerServiceType.GetMethod(GetDiagnosticsForSpanAsyncMethodName);
+			_getDiagnosticOnTextSpanMethod.ThrowOnNull(nameof(_getDiagnosticOnTextSpanMethod));
+
+			_taskResultPropertyInfo = GetTaskResultPropertyInfo(DiagnosticDataType);
+			_taskResultPropertyInfo.ThrowOnNull(nameof(_taskResultPropertyInfo));
+		}
+
+		private static Type GetInternalRoslynServiceType()
+		{
+			Type diagnosticAnalyzerServiceType = (from assembly in AppDomain.CurrentDomain.GetAssemblies()
+												  where assembly.FullName.StartsWith(RoslynDiagnosticServiceAssemblyName)
+												  from type in assembly.GetTypes()
+												  where type.IsInterface && type.Name == RoslynDiagnosticServiceTypeName
+												  select type)
+												 .FirstOrDefault();
+							  
+			return diagnosticAnalyzerServiceType;
+		}
+
+		private static Type GetDiagnosticDataType() =>
+			(from type in typeof(DocumentId).Assembly.GetTypes()
+			 where type.Name == DiagnosticDataTypeName
+			 select type)
+			.FirstOrDefault();
+
+		private static object GetDiagnosticServiceInstance(IComponentModel componentModel, Type diagnosticAnalyzerServiceType)
+		{
+			Type componentModelType = componentModel.GetType();
+			var getServiceMethodInfo = componentModelType.GetMethod(nameof(componentModel.GetService))
```

Comment:
Relying on Microsoft's internal implementation is very fragile

Reply:
Yes, but we don't have a choice unless they make this internal service public. There is no other way. And the access to internal service gives us huge benefit:
- the ability to add diagnostic to suppression
- the ability to write "Report false alert" command in the future

Reply:
Actually may be we can add some unit tests for it in the future. To check that Microsoft didn't change the implementation